src/liballoc/collections/btree/set.rs

```diff
@@ -544,42 +603,50 @@ impl<T: Ord> BTreeSet<T> {
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn is_subset(&self, other: &BTreeSet<T>) -> bool {
         // Same result as self.difference(other).next().is_none()
-        // but the 3 paths below are faster (in order: hugely, 20%, 5%).
-        if self.len() > other.len() {
-            false
-        } else if self.len() > other.len() / ITER_PERFORMANCE_TIPPING_SIZE_DIFF {
-            // Self is not much smaller than other set.
-            // Stolen from TreeMap
-            let mut x = self.iter();
-            let mut y = other.iter();
-            let mut a = x.next();
-            let mut b = y.next();
-            while a.is_some() {
-                if b.is_none() {
-                    return false;
-                }
-
-                let a1 = a.unwrap();
-                let b1 = b.unwrap();
-
-                match b1.cmp(a1) {
-                    Less => (),
-                    Greater => return false,
-                    Equal => a = x.next(),
-                }
-
-                b = y.next();
-            }
-            true
-        } else {
-            // Big difference in number of elements, or both sets are empty.
-            // Iterate the small set, searching for matches in the large set.
-            for next in self {
-                if !other.contains(next) {
-                    return false;
+        // but the code below is faster (hugely in some cases).
+        match self.len() {
+            larger if larger > other.len() => false,
+            0 => true,
+            1 => other.contains(self.iter().next().unwrap()),
+            _ => {
+                let mut self_except_first = self.iter();
+                let self_min = self_except_first.next().unwrap();
+                let mut self_except_last = self.iter();
+                let self_max = self_except_last.next_back().unwrap();
+                let mut other_iter = other.iter();
+                let other_min = other_iter.next().unwrap();
+                let other_max = other_iter.next_back().unwrap();
+                let mut self_iter = match (self_min.cmp(other_min), self_max.cmp(other_max)) {
```

Comment:
In the previous method you use the `Ord::cmp(x, y)` style and here `x.cmp(y)`. Either is fine but consistency is best.

Reply:
I never noticed that. Let's count: we have 4 x Ord::cmp, 3 x cmp (counting pairs as one). Before I started messing about in this code, there was just 1 Ord::cmp and 2 cmp. Notice that `cmp_opt` acts as a replacement for `Ord::cmp` but uses cmp itself.

So I say, use the shorter, member cmp.